src/hooks/transactions/updater.tsx

```diff
@@ -79,6 +79,7 @@ export default function Updater({ pendingTransactions, onCheck, onReceipt }: Upd
         promise
           .then((receipt) => {
             if (receipt) {
+              fastForwardBlockNumber(receipt.blockNumber)
```

Comment:
not sure why this is necessary

Reply:
This guarantees that the token allowance (`useTokenAllowance`) is synced to at least the block where it was confirmed. Without this, its possible that `useBlockNumber` returns a block prior to the approval's mining, in which case the UI will re-prompt for approval.

It's a bug in our block number / transaction context, which is fixed by fast forwarding in the case that a confirmation is observed in a block after the current block number.

Reply:
the `fastForward` existed before though. Where was it being called and why do we need to also add it here? Can we consolidate?

Reply:
It _wasn't_ being called, which was a benign bug which became malignant with this change.

Reply:
oh wow lol